heartbeat/osceip

```diff
@@ -0,0 +1,277 @@
+
+: ${OCF_RESKEY_osccli=${OCF_RESKEY_osccli_default}}
+: ${OCF_RESKEY_profile=${OCF_RESKEY_profile_default}}
+: ${OCF_RESKEY_region=${OCF_RESKEY_region_default}}
+
+meta_data() {
+    cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="osceip" version="1.0">
+<version>1.0</version>
+
+<longdesc lang="en">
+Resource Agent for Outscale Elastic IP Addresses.
+
+It manages Outscale Elastic IP Addresses with osc-cli.
+
+Credentials needs to be setup.
+
+See https://docs.outscale.com/en/userguide/Installing-and-Configuring-OSC-CLI.html for more information about osccli.
+</longdesc>
+<shortdesc lang="en">Outscale Elastic IP Address Resource Agent</shortdesc>
+
+<parameters>
+
+<parameter name="osccli" unique="0">
+<longdesc lang="en">
+command line tools for outscale services
+</longdesc>
+<shortdesc lang="en">OSC-CLI tools</shortdesc>
+<content type="string" default="${OCF_RESKEY_osccli_default}" />
+</parameter>
+
+<parameter name="profile">
+<longdesc lang="en">
+Valid OSC-CLI profile name (see ~/.osc/config and 'osc configure')
+</longdesc>
+<shortdesc lang="en">profile name</shortdesc>
+<content type="string" default="${OCF_RESKEY_profile_default}" />
+</parameter>
+
+<parameter name="elastic_ip" unique="1" required="1">
+<longdesc lang="en">
+reserved elastic ip for vm instance
+</longdesc>
+<shortdesc lang="en">reserved elastic ip for vm instance</shortdesc>
+<content type="string" default="" />
+</parameter>
+
+<parameter name="region" required="0">
+<longdesc lang="en">
+Region for osc resource (required for role-based authentication)
+</longdesc>
+<shortdesc lang="en">Region</shortdesc>
+<content type="string" default="${OCF_RESKEY_region_default}" />
+</parameter>
+
+<parameter name="curl_retries" unique="0">
+<longdesc lang="en">
+curl retries before failing
+</longdesc>
+<shortdesc lang="en">curl retries</shortdesc>
+<content type="integer" default="${OCF_RESKEY_curl_retries_default}" />
+</parameter>
+
+<parameter name="curl_sleep" unique="0">
+<longdesc lang="en">
+curl sleep between tries
+</longdesc>
+<shortdesc lang="en">curl sleep</shortdesc>
+<content type="integer" default="${OCF_RESKEY_curl_sleep_default}" />
+</parameter>
+
+
+</parameters>
+
+<actions>
+<action name="start"        timeout="30s" />
+<action name="stop"         timeout="30s" />
+<action name="monitor"      timeout="30s" interval="20s" depth="0" />
+<action name="migrate_to"   timeout="30s" />
+<action name="migrate_from" timeout="30s" />
+<action name="meta-data"    timeout="5s" />
+<action name="validate"     timeout="10s" />
+<action name="validate-all" timeout="10s" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+osceip_usage() {
+    cat <<END
+usage: $0 {start|stop|monitor|migrate_to|migrate_from|validate|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+curl_retry()
```

Comment:
We dont want curl_retry() here, as it's in ocf-shellfuncs.

Reply:
Unable to use ocf-shellfunc's curl_retry (curl_retry not found). I tested the awseip agent which is the only agent using it and I have the same problem. This problem appears when using the agent and not in standalone mode. I am on an ubuntu 22.04

Reply:
May 22 07:49:00.915 hap-b.test.local pacemaker-execd     [733] (log_op_output)  notice: elastic_stop_0[8386] error output [ /usr/lib/ocf/resource.d/heartbeat/osceip: 220: curl_retry: not found ]

Reply:
Ah, yeah. You'll have to patch /usr/lib/ocf/lib/heartbeat/ocf-shellfuncs in that case.

You should be able to do it by running the following commands:
```
cd  /usr/lib/ocf/lib/heartbeat
curl -LO https://github.com/ClusterLabs/resource-agents/pull/1936/commits/fc0657b936f6a58f741e33f851b22f82bc68bffa.patch
patch -p2 -F2 < fc0657b936f6a58f741e33f851b22f82bc68bffa.patch
rm fc0657b936f6a58f741e33f851b22f82bc68bffa.patch
```

Reply:
Is it possible to install resource agents from source ?

Reply:
Yes, it is.

You might need to install some -dev packages for some libraries, which it will complain about are missing when you run ./configure:
```
./autogen.sh

# rerun after installing lib<name>-dev it was complaining about until it completes without errors
./configure
make
make install
```

Reply:
fixed